pkg/middlewares/ratelimiter/redisrate/rate.go

```diff
@@ -0,0 +1,151 @@
+package redisrate
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+	"golang.org/x/time/rate"
+)
+
+const redisPrefix = "rate:"
+
+// Inf is the infinite rate limit; it allows all events (even if burst is zero).
+const Inf = rate.Limit(math.MaxFloat64)
+
+// InfDuration is the duration returned by Delay when a Reservation is not OK.
+const InfDuration = time.Duration(math.MaxInt64)
+
+type rediser interface {
+	Eval(ctx context.Context, script string, keys []string, args ...interface{}) *redis.Cmd
+	EvalSha(ctx context.Context, sha1 string, keys []string, args ...interface{}) *redis.Cmd
+	ScriptExists(ctx context.Context, hashes ...string) *redis.BoolSliceCmd
+	ScriptLoad(ctx context.Context, script string) *redis.StringCmd
+	Del(ctx context.Context, keys ...string) *redis.IntCmd
+
+	EvalRO(ctx context.Context, script string, keys []string, args ...interface{}) *redis.Cmd
+	EvalShaRO(ctx context.Context, sha1 string, keys []string, args ...interface{}) *redis.Cmd
+}
+
+type Limit struct {
+	Rate   rate.Limit
+	Burst  int64
+	Period time.Duration
+}
+
+func (l Limit) String() string {
+	return fmt.Sprintf("%f req/%s (burst %d)", l.Rate, fmtDur(l.Period), l.Burst)
```

Comment:
The rate is already expressed in req/s. So, I think the `fmtDur` call here will generate a wrong message.
I didn't see any use of `String`, `IsZero`, `PerSecond`, `PerMinute`, `PerHour`, could you delete them?